unity/Holo/Assets/Scripts/STLConverter.cs

```diff
@@ -0,0 +1,178 @@
+using UnityEngine;
+using UnityEditor;
+using System.Text;
+using System.Collections.Generic;
+using System.IO;
+
+/// A class for importing and converting StL series into an animation
+public class STLSeriesConverter
+{
+    GameObject series_GameObject;
+    
+    [MenuItem("Holo/Convert STL series to a .prefab")]
+    public static void ConvertSTL()
+    {
+        STLSeriesConverter stlSeriesConverter = new STLSeriesConverter();
+        STLSeriesImporter stlImporter = new STLSeriesImporter();
+
+        stlSeriesConverter.series_GameObject = stlImporter.Get_GameObject();
+        stlSeriesConverter.series_GameObject.AddComponent<BlendShapeAnimation>();
+
+        stlSeriesConverter.ExportToPrefab();
+    }
+
+    // Exports finished GameObject to a .prefab
+    private void ExportToPrefab()
+    {
+        //string save_path = EditorUtility.SaveFilePanel("Choose a folder where for a .prefab file", "", "", ".prefab");
+        string save_path = @"C: \Users\mit - kuchnowski\repos\holo\unity\Holo\Assets\Resources\test2.prefab";
+        PrefabUtility.SaveAsPrefabAsset(series_GameObject, save_path);
+    }
+
+}
+
+//A class for importing a ASTL series from a dir
+public class STLSeriesImporter
+{
+    GameObject imported_STLs = new GameObject("Hypertrophy");
+
+    private string[] file_paths;
+
+    public GameObject Get_GameObject()
+    {
+        return this.imported_STLs;
+    }
+
+    // Object constructor, initiates STL series import.
+    public STLSeriesImporter()
+    {
+        this.Get_file_paths();
+        this.Load_files();
+    }
+
+
+    // gets path to the subsequent STL meshes stored in a root folder.
+    private void Get_file_paths()
+    {
+        //string root_folder = EditorUtility.OpenFolderPanel("Select STL series root folder", "", "");
+        string root_folder = @"C:\Users\mit-kuchnowski\repos\holo\data\hypertrophy-heart\meshes\";
+        file_paths = Directory.GetFiles(root_folder);
+    }
+
+
+    //loads meshes from separate files into one GameObject
+    private void Load_files()
+    {
+        SkinnedMeshRenderer skinnedMesh = imported_STLs.AddComponent<SkinnedMeshRenderer>();
+        Mesh mesh = new Mesh();
+
+        STLFileImporter stlFileImporter = new STLFileImporter();
+
+        bool first_mesh = true;
+        Debug.Log("Doing mesh #1!");
+        stlFileImporter.Load_STL_file(file_paths[0], first_mesh);
+        mesh.vertices = stlFileImporter.BaseVertices;
+        mesh.triangles = stlFileImporter.Indices;
+        mesh.AddBlendShapeFrame(Path.GetFileName(file_paths[0]), 100f, stlFileImporter.AllVertices, null, null);
+        first_mesh = false;
+
+        int count = 2;
+        for (int i = 1; i < file_paths.Length; i++)
+        {
+            Debug.Log("Doing mesh #" + count.ToString() + "!");
+            stlFileImporter.Load_STL_file(file_paths[1], first_mesh);
+            mesh.AddBlendShapeFrame(Path.GetFileName(file_paths[i]), 100f, stlFileImporter.AllVertices, null, null);
```

Comment:
Here you can check that all STLs have equal topology. The `stlFileImporter.AllVertices.Length` and `stlFileImporter.Indices` at this point should be equal to values that were returned after `.Load_STL_file(file_paths[0], first_mesh)`.